### PR TITLE
wasm_runtime_start_debug_instance: Allow to override port

### PR DESCRIPTION
**Pull request by yamt**
This allows the embedder to manage port numbers for this purpose
by itself.

### CODE REVIEW

File: product-mini/platforms/posix/main.c [hunk 1]
```diff
@@ -577,7 +577,7 @@ main(int argc, char *argv[])
             printf("%s\n", wasm_runtime_get_exception(wasm_module_inst));
             goto fail4;
         }
-        debug_port = wasm_runtime_start_debug_instance(exec_env);
+        debug_port = wasm_runtime_start_debug_instance(exec_env, -1);
```
Comment (xujuntwt95329) [inline product-mini/platforms/posix/main.c:580]: Don't need to change this

File: product-mini/platforms/windows/main.c [hunk 2]
```diff
@@ -444,7 +444,7 @@ main(int argc, char *argv[])
             printf("%s\n", wasm_runtime_get_exception(wasm_module_inst));
             goto fail4;
         }
-        debug_port = wasm_runtime_start_debug_instance(exec_env);
+        debug_port = wasm_runtime_start_debug_instance(exec_env, -1);
```
Comment (xujuntwt95329) [inline product-mini/platforms/windows/main.c:447]: Same as above

File: core/iwasm/common/wasm_runtime_common.c [hunk 3]
```diff
@@ -507,7 +507,7 @@ wasm_runtime_is_xip_file(const uint8 *buf, uint32 size)
 
 #if (WASM_ENABLE_THREAD_MGR != 0) && (WASM_ENABLE_DEBUG_INTERP != 0)
 uint32
-wasm_runtime_start_debug_instance(WASMExecEnv *exec_env)
+wasm_runtime_start_debug_instance(WASMExecEnv *exec_env, int32_t port)
```
Comment (xujuntwt95329) [inline core/iwasm/common/wasm_runtime_common.c:510]: This is a public API and has been used in some other codes.

Had better remain this API unchanged and add a new API `wasm_runtime_start_debug_instance_with_port`, then `wasm_runtime_start_debug_instance` can call `wasm_runtime_start_debug_instance_with_port` with port = -1

Comment (yamt) [reply core/iwasm/common/wasm_runtime_common.c]: it makes sense. thank you. done.

File: product-mini/platforms/windows/main.c (hunk 2 above)
Comment (yamt) [reply product-mini/platforms/windows/main.c]: done

File: product-mini/platforms/posix/main.c (hunk 1 above)
Comment (yamt) [reply product-mini/platforms/posix/main.c]: done

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Pull request merged**